assets/apps/calendar/init.js

```diff
@@ -127,32 +134,83 @@ window.addEventListener('message', function (e) {
     } else if (e.data.type == "loadAdditional") {
         loadAdditional(e.data.currentMonth, e.data.yearMonth);
     } else if (e.data.type == "respondRenameCalendar") {
-        respondToCalendarRename(e.data.oldName, e.data.newName);
+        respondToCalendarRename(e.data.calendar);
+    } else if (e.data.type == "respondDeleteCalendar") {
+        respondToCalendarDelete(e.data.calendar);
+    } else if (e.data.type == "respondAddCalendar") {
+        respondToCalendarAdd(e.data.newId, e.data.newName, e.data.newColor);
+    } else if (e.data.type == "respondCalendarColorChange") {
+        respondToCalendarColorChange(e.data.calendarName, e.data.newColor);
     } else if(e.data.type == "importICSFile") {
         loadCalendarAsGuest = e.data.loadCalendarAsGuest;
         if(loadCalendarAsGuest) {
             initialiseCalendar(true, []);
         } else {
             setCalendars(true, []);
         }
-        importICSFile(e.data.contents, e.data.username, e.data.isSharedWithUs, loadCalendarAsGuest);
+        importICSFile(e.data.contents, e.data.username, e.data.isSharedWithUs, loadCalendarAsGuest, "default", true);
     }
 });
-function renameCalendar(currentName) {
-    mainWindow.postMessage({action:'requestRenameCalendar', currentName: currentName}, origin);
+function renameCalendar(calendar) {
+    mainWindow.postMessage({action:'requestRenameCalendar', calendar: calendar}, origin);
+}
+function addCalendar(){
+    let idAsInt = new Number(CALENDAR_ID_MY_CALENDAR) + CalendarList.length;
+    let color = colorPalette[(idAsInt % 8) + 8];
+    mainWindow.postMessage({action:'requestAddCalendar', newColor:color}, origin);
```

Comment:
why do these need to go back to the trusted code? can't we implement them in the app directly?

Reply:
I wanted to use the existing dialogs for add, rename. sandboxed apps would use there own.

Reply:
So long as we're aware we'll need to pull them out once we have a sandbox.